doc/whats_new/v0.23.rst

```diff
@@ -389,6 +389,14 @@ Changelog
   :class:`tree.DecisionTreeRegressor`. :pr:`15864` by
   `Nicolas Hug`_.
 
+:mod:`sklearn.impute`
+...............................
+
+- |Enhancement| :class:`impute.IterativeImputer` Will set the estimator's
+  `random_state` to the same `random_state` that was passed to the imputer
+  instead of a `RandomState`, allowing to use it with more external classes.
+  :pr:`15636` by :user:`David Cortes <david-cortes>`.
```

Comment:
This needs to move to 0.24 now.

Reply:
Not sure what you mean by current approach with 'int'. The current approach (as it is in the master branch right now) is to assign a `RandomState` object to the estimator’s `random_state` if it has such attribute.

Reply:
Third party estimators often restrict `random_state` to only be integers. In the current state of this PR, we are passing the integer to `self._estimators`, which I am okay with.

Reply:
If a user were to use a `RandomState` object with an `self._estimator` that does not support `RandomState` objects, this will fail.

Reply:
Moved the whatsnew notice to 0.24 and turned the test into an equality test of the `RandomState` states.